src/tools/crawler/crawl.rs

```diff
@@ -10,70 +11,170 @@ use serde::Deserialize;
 use thiserror::Error;
 use tokio::time::Instant;
 
-#[allow(dead_code)]
-#[derive(Debug, Deserialize)]
-pub(super) struct Peer {
-    pub(super) complete_ledgers: Option<String>,
-    pub(super) complete_shards: Option<String>,
-    pub(super) ip: Option<String>,
-    pub(super) port: Option<Port>,
-    pub(super) public_key: String,
+/// Each member of the overlay.active array is an object with the following fields.
+#[derive(Debug, Deserialize, Clone)]
+pub struct Peer {
+    /// The range of ledger versions this peer has available.
+    pub complete_ledgers: Option<String>,
+
+    /// The range of ledger history shards this peer has available.
+    pub complete_shards: Option<String>,
+
+    /// The IP address of this connected peer.
+    ///
+    /// Omitted if the peer is configured as a validator or a private peer.
+    pub ip: Option<String>,
+
+    /// The port number on the peer server that serves RTXP.
+    ///
+    /// Typically 51235.
+    /// Omitted if the peer is configured as a validator or a private peer.
+    pub port: Option<Port>,
+
+    /// The public key of the ECDSA key pair used by this peer to sign RTXP messages.
+    pub public_key: String,
+
+    /// Indicating whether the TCP connection to the peer is incoming or outgoing.
+    ///
+    /// The value is "in" or "out".
     #[serde(rename = "type")]
-    pub(super) connection_type: String,
-    pub(super) uptime: u32,
-    pub(super) version: String,
+    pub connection_type: String,
```

Comment:
I'm not aware about the source of this information (maybe it's somewhere else also string and then it's fine to have string here), but if that field can have only two possible values wouldn't enum be more appropriate?